lib/preprocessor.py

```diff
@@ -388,10 +388,13 @@ def filter(self, ctype, ref, ret):
                 "Wrong command for filters: {0}".format(repr(ctype)))
 
     def lua_eval(self, name, expr, silent=True):
-        self.servers[name].admin.reconnect()
-        result = self.servers[name].admin(
-            '%s%s' % (expr, self.delimiter), silent=silent
-        )
+        try:
+            self.servers[name].admin.reconnect()
+            result = self.servers[name].admin(
+                '%s%s' % (expr, self.delimiter), silent=silent
+            )
+        except KeyError:
+            return []
```

Comment:
The same as for PR #228: we should not silently pass over the incorrect command: attempt to evaluate a command on a nonexistent server in the case. The correct behaviour would be:

1. Write a notice to the screen.
2. Don't stop a test.
3. Raise a Lua error in the test code that calls `test_run:eval()`.

I have implemented it this way in the first two commits in PR #237 ('Handle incorrect test-run commands gracefully', 'Give meaningful error messages for failed commands').